s3cloudfront/s3-cloudfront.tf

```diff
@@ -0,0 +1,141 @@
+terraform {
+  backend "s3" {
+    bucket = "fp-${fp_context}-cdn-terraform-state"
+    region = "${AWS_DEFAULT_REGION}"
+    key    = "${fp_context}-cdn.tfstate"
+  }
+}
+
+# Create S3 bucket.
+resource "aws_s3_bucket" "cdn" {
+    bucket = "${fp_context}-cdn-bucket"
+    tags = {
+       Name = "${var.fp_context}-cdn-bucket"
+     } 
+}
+
+# Create Origin Access Identity for CDN-S3.
+resource "aws_cloudfront_origin_access_identity" "oai" {
+    comment = "cloudfront origin access identity"
+}
+
+locals {
+  s3_origin_id = "org_${aws_s3_bucket.cdn}"
+}
+# origin - cloudfront get its content.
+resource "aws_cloudfront_distribution" "fp_cdn" {
+  origin {
+    domain_name = aws_s3_bucket.cdn.bucket_regional_domain_name
+    origin_id   = local.s3_origin_id
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.oai.cloudfront_access_identity_path
+    }
+  }
+
+  enabled             = true
+  is_ipv6_enabled     = true
+  comment             = "Fight Pandemic CDN distribution"
+  default_root_object = "index.html"
+ 
+ # logging cofig 
+ /*
+  logging_config {
+    include_cookies = false
+    bucket          = "mylogs.s3.amazonaws.com"
```

Comment:
Let's store the logs in an S3 bucket called `cdn-logs.s3.amazonaws.com`. Since the this bucket will only store the CDN access logs, I don't think we need a prefix.